### PR TITLE
#9426 Self sizing header for dynamic type in firefox homepage

### PR DESCRIPTION
**Pull request by lmarceau**
- Pull requests to start fixing https://github.com/mozilla-mobile/firefox-ios/issues/9426
- This fixes the header on the firefox home page so it can adjust to dynamic type (so header doesn't have a fixed height anymore)
- Added a function to get the preferred bold font (was only possible to get the standard font)
- Fixed some typos at the same time

### CODE REVIEW

**Comment by lmarceau** [comment]
What it looks like. Please note other section are not fixed yet, only the header is.

iPhone 6s:
Simulator Screen Shot - iPhone 6s - 2021-11-01 at 12 31 13

iPad:
Simulator Screen Shot - iPad Pro (12 9-inch) (5th generation) - 2021-11-01 at 12 28 43

File: Client/Frontend/Home/FirefoxHomeViewController.swift [hunk 1]
```diff
@@ -707,25 +714,26 @@ extension FirefoxHomeViewController: UICollectionViewDelegateFlowLayout {
...
+            return isTopSitesSectionEnabled ? getHeaderSize(forSection: section) : .zero
         case .libraryShortcuts:
-            return isYourLibrarySectionEnabled ? Section(section).headerHeight : .zero
+            return isYourLibrarySectionEnabled ? getHeaderSize(forSection: section) : .zero
         case .jumpBackIn:
-            return isJumpBackInSectionEnabled ? Section(section).headerHeight : .zero
+            return isJumpBackInSectionEnabled ? getHeaderSize(forSection: section) : .zero
         case .recentlySaved:
-            return isRecentlySavedSectionEnabled ? Section(section).headerHeight : .zero
+            return isRecentlySavedSectionEnabled ? getHeaderSize(forSection: section) : .zero
```
Comment (electricRGB) [inline Client/Frontend/Home/FirefoxHomeViewController.swift:736]: This is great! I would also double check: are we using `Section(section).headerHeight` anywhere else? If not, can we get rid of it?

**Review by electricRGB — APPROVED** [review]

File: Client/Frontend/Home/FirefoxHomeViewController.swift (hunk 1 above)
Comment (lmarceau) [reply Client/Frontend/Home/FirefoxHomeViewController.swift:736]: Yessir, we can get rid of it! It's in the PR 😇
Screen Shot 2021-11-01 at 4 56 18 PM

Comment (electricRGB) [reply Client/Frontend/Home/FirefoxHomeViewController.swift:736]: I obviously did not fully pay attention and forgot. Foot in mouth!

**Comment by lmarceau** [comment]
The two failing smoke tests passes locally:
- XCUITests.SearchTests testSearchEngine
Screen Shot 2021-11-02 at 10 42 06 AM

- XCUITests.AuthenticationTest testBasicHTTPAuthenticationPromptVisible
Screen Shot 2021-11-02 at 10 27 35 AM

**Pull request merged**